From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: zml <zml@stellardrift.ca>
Date: Sat, 20 Mar 2021 21:30:47 -0700
Subject: [PATCH] emitter: Handle comments being written in flow nodes, so at
 least we don't crash


diff --git a/src/main/java/org/yaml/snakeyaml/emitter/Emitter.java b/src/main/java/org/yaml/snakeyaml/emitter/Emitter.java
index fa5fa82564e9b24d36ab59052e614b80dd570487..b2d846ca8663b482d0b5ed19594cb2b22f4b5957 100644
--- a/src/main/java/org/yaml/snakeyaml/emitter/Emitter.java
+++ b/src/main/java/org/yaml/snakeyaml/emitter/Emitter.java
@@ -547,7 +547,7 @@ public final class Emitter implements Emitable {
                 }
                 state = states.pop();
             } else if (event instanceof CommentEvent) {
-                blockCommentsCollector.collectEvents(event);
+                event = blockCommentsCollector.collectEvents(event);
                 writeBlockComment();
             } else {
                 writeIndicator(",", false, false, false);
@@ -556,7 +556,7 @@ public final class Emitter implements Emitable {
                 }
                 states.push(new ExpectFlowSequenceItem());
                 expectNode(false, false, false);
-                inlineCommentsCollector.collectEvents(event);
+                event = inlineCommentsCollector.collectEvents(event);
                 writeInlineComments();
             }
         }
@@ -587,6 +587,8 @@ public final class Emitter implements Emitable {
                 if (canonical || (column > bestWidth && splitLines) || prettyFlow) {
                     writeIndent();
                 }
+                event = blockCommentsCollector.collectEventsAndPoll(event);
+                writeBlockComment();
                 if (!canonical && checkSimpleKey()) {
                     states.push(new ExpectFlowMappingSimpleValue());
                     expectNode(false, true, true);
@@ -617,6 +619,8 @@ public final class Emitter implements Emitable {
                 state = states.pop();
             } else {
                 writeIndicator(",", false, false, false);
+                event = blockCommentsCollector.collectEventsAndPoll(event);
+                writeBlockComment();
                 if (canonical || (column > bestWidth && splitLines) || prettyFlow) {
                     writeIndent();
                 }
