From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: zml <zml@stellardrift.ca>
Date: Sat, 7 Aug 2021 19:58:26 -0700
Subject: [PATCH] Make prettyFlow modifiable during emission

This allows controlling prettyFlow on a per-node basis.

diff --git a/src/main/java/org/yaml/snakeyaml/emitter/Emitter.java b/src/main/java/org/yaml/snakeyaml/emitter/Emitter.java
index cad49156be009cd37eeb0bb0bd70248908ef1630..504d04746ec0952f65d93e2b2137a78c03815f29 100644
--- a/src/main/java/org/yaml/snakeyaml/emitter/Emitter.java
+++ b/src/main/java/org/yaml/snakeyaml/emitter/Emitter.java
@@ -151,7 +151,7 @@ public final class Emitter implements Emitable {
     // Formatting details.
     private final Boolean canonical;
     // pretty print flow by adding extra line breaks
-    private final Boolean prettyFlow;
+    private boolean prettyFlow; // Configurate - non-final
 
     private final boolean allowUnicode;
     private int bestIndent;
@@ -248,6 +248,16 @@ public final class Emitter implements Emitable {
                 CommentType.IN_LINE);
     }
 
+    // Configurate start - customizable prettyFlow
+    public void prettyFlow(final boolean prettyFlow) {
+        this.prettyFlow = prettyFlow;
+    }
+
+    public boolean prettyFlow() {
+        return this.prettyFlow;
+    }
+    // Configurate end
+
     public void emit(Event event) throws IOException {
         this.events.add(event);
         while (!needMoreEvents()) {
